tw-tasks-core/src/main/java/com/transferwise/tasks/processing/TasksProcessingService.java

```diff
@@ -213,8 +213,7 @@ protected ProcessTasksResponse processTasks(GlobalProcessingState.Bucket bucket)
           mdcService.put(task);
           try {
             ProcessTaskResponse processTaskResponse = grabTaskForProcessing(bucketId, task);
-
-            coreMetricsTemplate.registerTaskGrabbingResponse(bucketId, type, priority, processTaskResponse);
+            coreMetricsTemplate.registerTaskGrabbingResponse(bucketId, type, priority, processTaskResponse, taskTriggering.getTriggerAt());
```

Comment:
Will this cause a NPE with old version of `taskTriggering` DTO? One that has no `triggeredAt` saved yet.

Reply:
Looks like `TaskTriggering` is initialised in two places and we set the `triggeredAt` with both of them.

Kafka one comes from a `long` from the event timestamp and other one comes from `Instant.now()` so think should be fine.

Would prefer to make triggeredAt final and force its initiation during constructor but seems the library exclusively uses init then set pattern so followed this

https://github.com/transferwise/tw-tasks-executor/blob/e995b032612f996f5ff47757e4be43e73f0957ac/tw-tasks-core/src/main/java/com/transferwise/tasks/triggering/KafkaTasksExecutionTriggerer.java#L184-L187

https://github.com/transferwise/tw-tasks-executor/blob/e995b032612f996f5ff47757e4be43e73f0957ac/tw-tasks-core/src/main/java/com/transferwise/tasks/triggering/KafkaTasksExecutionTriggerer.java#L291-L296